RSSDP/SsdpDevicePublisher.cs

```diff
@@ -472,7 +492,12 @@ private Task SendByeByeNotification(SsdpDevice device, string notificationType,
 
             var sendCount = IsDisposed ? 1 : 3;
             WriteTrace(String.Format("Sent byebye notification"), device);
-            return _CommsServer.SendMulticastMessage(message, sendCount, cancellationToken);
+            if (_sendOnlyMatchedHost)
+            {
+                return _CommsServer.SendMulticastMessage(message, sendCount, device.ToRootDevice().Address, cancellationToken);
```

Comment:
same, ternary could fit IMHO
```suggestion
                return _CommsServer.SendMulticastMessage(message, sendCount,
                        _sendOnlyMatchedHost ? device.ToRootDevice().Address : null, cancellationToken);
```